### DISCUSSION TITLE
How is "Toggle conceallevel" supposed to work?

### DISCUSSION THREAD

**Discussion by LorenzoBettini** [LazyVim/LazyVim#4986] [Q&A] [answered]
I've enabled markdown extras and markdown elements are "rendered" thanks to the default conceallevel, and that's fine!
However, `<leader>uc` does not toggle conceallevel: I always get the popup saying "Enabled conceallevel", no matter how many times I use that keybinding.
Interestingly, if I do `<leader>sk` to search for keybindings and I select "Toggle conceallevel" from there, I finally get "Disabled conceallevel", but markdown elements are still "hidden", so it looks like the conceallevel always stays the same...

am I missing something?

thanks in advance

**Comment by dpetka2001** [comment, marked as answer]
You also have to toggle `render-markdown` off. By default it always renders with `conceallevel = 3` (see [plugin's official docs here](https://github.com/MeanderingProgrammer/render-markdown.nvim/blob/e41b0002fe4196825450ab5a6343300c40791d51/README.md?plain=1#L639-L644)). When you toggle it off with `<leader>um`, then you can also use correctly `<leader>uc` to toggle `conceallevel` between 2 and 0 which are LazyVim's options for toggling the `conceallevel`.

**Reply by LorenzoBettini** [reply]
Ah! You're right! Thanks! :)
That isn't very clear though...

**Reply by dpetka2001** [reply]
Well you have to read the plugin's docs to find out about it. It's what the plugin does on its own (setting `conceallevel = 3` always when it's rendering the page). LazyVim docs can't have any single detail about what plugins do in their configuration.

**Reply by vaibhav-kaushal** [reply]
> You also have to toggle `render-markdown` off. By default it always renders with `conceallevel = 3` (see [plugin's official docs here](https://github.com/MeanderingProgrammer/render-markdown.nvim/blob/e41b0002fe4196825450ab5a6343300c40791d51/README.md?plain=1#L639-L644)). When you toggle it off with `<leader>um`, then you can also use correctly `<leader>uc` to toggle `conceallevel` between 2 and 0 which are LazyVim's options for toggling the `conceallevel`.

Is it possible to set that behavior in the config?

I am sorry if this is a noob question.

**Reply by cody-vibe** [reply]
@vaibhav-kaushal I have the following in my plugin config:

```lua
return {
  {
    'MeanderingProgrammer/render-markdown.nvim',
    opts = {
      enabled = true,
      render_modes = { 'n', 'c', 't', 'i' },
      link = {
        enabled = false
      },
      win_options = { conceallevel = { rendered = 0 } }
    }
  }
}
```
You need `enabled = true` so the plugin is active to be able to apply the `win_options`. Also I disable links to remove the icons.

This way my tables with links look exactly like they are when editing, and I can space align.
